### DISCUSSION TITLE
An alternative way to check the coverage report

### DISCUSSION THREAD

**Discussion by fktn-k** [fktn-k/fkYAML#297] [Ideas]
Currently, the coverage results for each library source file cannot be accessed in the Coveralls site for some reason, reported in the PR #293.  
As @stephenwhittle has pointed out, an alternative way to view a failed report would be a usability improvement, for example, for a one-off contributor, even if the above issue will have been fixed.  

To resolve the issue, I'm thinking of modifying the CI job which uploads coverage data to upload an HTML view generated from the uploaded coverage data as an artifact as well with some comments on the associated PR.  
Although the view is not the same as shown in the Coveralls site since the generators are different, there shouldn't be any difference in the converage data itself as far as I know.  
Maybe it's reasonable that the latest artifact overwrites the previous one for a PR so the storage space can be saved.  

What do you think about that, @stephenwhittle?

**Comment by stephenwhittle** [comment]
This would be good - especially as it guarantees that I as a contributor am actually seeing the coverage data that the CI job is generating, rather than generating it locally and hoping it's deterministic enough to get the same results. 
It's probably reasonable to just overwrite the last artifact for a given PR, yes.

**Reply by fktn-k** [reply]
Thanks. I'll give it a try in the PR #295.

**Reply by fktn-k** [reply]
@stephenwhittle   
Finally I've added the functionality to the CI job, notifying the event using the github-actions bot.  
I think that's decent, but I want you to take a look #295 before merging the PR if possible.

**Reply by fktn-k** [reply]
@stephenwhittle   
I've merged the PR for incoming ones.  
If you find any issue or potential improvements, please let me know as usual.
